openapi/schemas/GetTimeRegistrationsByTaskRequest.yaml

```diff
@@ -0,0 +1,29 @@
+type: array
+items:
+  type: object
+  properties:
+    taskId:
+      type: string
+    taskName:
+      type: string
+    taskDescription:
+      type: string
+    timeRegistrations:
+      type: array
+      items:
+        type: "object"
+        required: [ "timeRegistrationId", "registered", "date", "duration", "status" ]
```

Comment:
I know you moved the `taskId` to the top level, since it's the dame for all. Will it make it harder to create clients that this type is unrelated to the existing `TimeRegistrationResponse`?

I'm not saying we should always use the same types across all endpoints, and it might be better to keep them separate for now.

Reply:
> I know you moved the taskId to the top level, since it's the dame for all. Will it make it harder to create clients that this type is unrelated to the existing `TimeRegistrationResponse`?

I think that will depend on how the client generates the types (if they do).

Generally for API stuff, I prefer keeping request/response types separate (even though they might be identical to another request/response type) as you never know if a future version will add or remove a property.